// @worldcoin/idkit/dist/index.cjs
'use strict';

var idkitCore = require('@worldcoin/idkit-core');
var react = require('react');
var reactDom = require('react-dom');
var jsxRuntime = require('react/jsx-runtime');
var QRCodeUtil = require('qrcode/lib/core/qrcode.js');

function _interopDefault (e) { return e && e.__esModule ? e : { default: e }; }

var QRCodeUtil__default = /*#__PURE__*/_interopDefault(QRCodeUtil);

// src/hooks/useIDKitRequest.ts
function createInitialHookState() {
  return {
    isOpen: false,
    status: "idle",
    connectorURI: null,
    result: null,
    errorCode: null
  };
}
function ensureNotAborted(signal) {
  if (signal?.aborted) {
    throw idkitCore.IDKitErrorCodes.Cancelled;
  }
}
async function delay(ms, signal) {
  if (!signal) {
    await new Promise((resolve) => setTimeout(resolve, ms));
    return;
  }
  await new Promise((resolve, reject) => {
    const timeout = setTimeout(() => {
      signal.removeEventListener("abort", abortHandler);
      resolve();
    }, ms);
    const abortHandler = () => {
      clearTimeout(timeout);
      signal.removeEventListener("abort", abortHandler);
      reject(idkitCore.IDKitErrorCodes.Cancelled);
    };
    signal.addEventListener("abort", abortHandler, { once: true });
  });
}
var knownErrorCodes = new Set(Object.values(idkitCore.IDKitErrorCodes));
function asKnownErrorCode(value) {
  if (typeof value === "string" && knownErrorCodes.has(value)) {
    return value;
  }
  return null;
}
function toErrorCode(error) {
  const directCode = asKnownErrorCode(error);
  if (directCode) {
    return directCode;
  }
  if (typeof error === "object" && error !== null && "code" in error) {
    const nestedCode = asKnownErrorCode(error.code);
    if (nestedCode) {
      return nestedCode;
    }
  }
  return idkitCore.IDKitErrorCodes.GenericError;
}

// src/hooks/useIDKitFlow.ts
function useIDKitFlow(createFlowHandle, config) {
  const [state, setState] = react.useState(
    createInitialHookState
  );
  const abortRef = react.useRef(null);
  const createFlowHandleRef = react.useRef(createFlowHandle);
  const configRef = react.useRef(config);
  createFlowHandleRef.current = createFlowHandle;
  configRef.current = config;
  const reset = react.useCallback(() => {
    abortRef.current?.abort();
    abortRef.current = null;
    setState(createInitialHookState);
  }, []);
  const open = react.useCallback(() => {
    setState((prev) => {
      if (prev.isOpen) {
        return prev;
      }
      return {
        isOpen: true,
        status: "waiting_for_connection",
        connectorURI: null,
        result: null,
        errorCode: null
      };
    });
  }, []);
  react.useEffect(() => {
    if (!state.isOpen) {
      return;
    }
    const controller = new AbortController();
    abortRef.current = controller;
    const setFailed = (errorCode) => {
      setState((prev) => {
        if (prev.status === "failed" && prev.errorCode === errorCode) {
          return prev;
        }
        return {
          ...prev,
          status: "failed",
          errorCode
        };
      });
    };
    void (async () => {
      try {
        const request = await createFlowHandleRef.current();
        ensureNotAborted(controller.signal);
        setState((prev) => {
          if (prev.connectorURI === request.connectorURI) {
            return prev;
          }
          return { ...prev, connectorURI: request.connectorURI };
        });
        const pollInterval = configRef.current.polling?.interval ?? 1e3;
        const timeout = configRef.current.polling?.timeout ?? 3e5;
        const startedAt = Date.now();
        while (true) {
          ensureNotAborted(controller.signal);
          if (Date.now() - startedAt > timeout) {
            setFailed(idkitCore.IDKitErrorCodes.Timeout);
            return;
          }
          const nextStatus = await request.pollOnce();
          ensureNotAborted(controller.signal);
          if (nextStatus.type === "confirmed") {
            const confirmedResult = nextStatus.result;
            if (!confirmedResult) {
              setFailed(idkitCore.IDKitErrorCodes.UnexpectedResponse);
              return;
            }
            setState((prev) => ({
              ...prev,
              status: "confirmed",
              result: confirmedResult,
              errorCode: null
            }));
            return;
          }
          if (nextStatus.type === "failed") {
            setFailed(nextStatus.error ?? idkitCore.IDKitErrorCodes.GenericError);
            return;
          }
          setState((prev) => {
            if (prev.status === nextStatus.type) {
              return prev;
            }
            return { ...prev, status: nextStatus.type };
          });
          await delay(pollInterval, controller.signal);
        }
      } catch (error) {
        if (controller.signal.aborted) {
          return;
        }
        setFailed(toErrorCode(error));
      }
    })();
    return () => {
      controller.abort();
      if (abortRef.current === controller) {
        abortRef.current = null;
      }
    };
  }, [state.isOpen]);
  return {
    open,
    reset,
    isAwaitingUserConnection: state.status === "waiting_for_connection",
    isAwaitingUserConfirmation: state.status === "awaiting_confirmation",
    isSuccess: state.status === "confirmed",
    isError: state.status === "failed",
    connectorURI: state.connectorURI,
    result: state.result,
    errorCode: state.errorCode,
    isOpen: state.isOpen
  };
}

// src/hooks/useIDKitRequest.ts
function useIDKitRequest(config) {
  return useIDKitFlow(
    () => idkitCore.IDKit.request({
      app_id: config.app_id,
      action: config.action,
      rp_context: config.rp_context,
      action_description: config.action_description,
      bridge_url: config.bridge_url,
      allow_legacy_proofs: config.allow_legacy_proofs,
      override_connect_base_url: config.override_connect_base_url,
      environment: config.environment
    }).preset(config.preset),
    config
  );
}

// src/styles/widgetStyles.ts
var WIDGET_STYLES = `
@font-face {
  font-family: 'TWK Lausanne';
  src: url('https://world-id-assets.com/fonts/TWKLausanne-200.woff2') format('woff2');
  font-weight: 200;
  font-style: normal;
  font-display: swap;
}
@font-face {
  font-family: 'TWK Lausanne';
  src: url('https://world-id-assets.com/fonts/TWKLausanne-300.woff2') format('woff2');
  font-weight: 300;
  font-style: normal;
  font-display: swap;
}
@font-face {
  font-family: 'TWK Lausanne';
  src: url('https://world-id-assets.com/fonts/TWKLausanne-400.woff2') format('woff2');
  font-weight: 400;
  font-style: normal;
  font-display: swap;
}
@font-face {
  font-family: 'TWK Lausanne';
  src: url('https://world-id-assets.com/fonts/TWKLausanne-500.woff2') format('woff2');
  font-weight: 500;
  font-style: normal;
  font-display: swap;
}
@font-face {
  font-family: 'TWK Lausanne';
  src: url('https://world-id-assets.com/fonts/TWKLausanne-600.woff2') format('woff2');
  font-weight: 600;
  font-style: normal;
  font-display: swap;
}

/* CSS Custom Properties */
:host {
  --idkit-font: 'TWK Lausanne', ui-sans-serif, system-ui, -apple-system, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif;
  --idkit-bg: #ffffff;
  --idkit-text: #0d151d;
  --idkit-text-muted: #657080;
  --idkit-text-secondary: #9eafc0;
  --idkit-border: #EBECEF;
  --idkit-border-light: #f1f5f8;
  --idkit-surface: #f8fafc;
  --idkit-success: #00C230;
  --idkit-error: #9BA3AE;
  --idkit-warning: #FFAE00;
  --idkit-btn-bg: #0d151d;
  --idkit-btn-text: #ffffff;
}
:host(.dark) {
  --idkit-bg: #0d151d;
  --idkit-text: #ffffff;
  --idkit-text-muted: #9eafc0;
  --idkit-text-secondary: #657080;
  --idkit-border: rgba(235, 236, 239, 0.15);
  --idkit-border-light: rgba(241, 245, 248, 0.1);
  --idkit-surface: rgba(255, 255, 255, 0.05);
  --idkit-btn-bg: #ffffff;
  --idkit-btn-text: #0d151d;
}

/* Animations */
@keyframes idkit-fade-in {
  from { opacity: 0; }
  to { opacity: 1; }
}
@keyframes idkit-scale-in {
  from { opacity: 0; transform: scale(0.9); }
  to { opacity: 1; transform: scale(1); }
}
@keyframes idkit-slide-up {
  from { transform: translateY(100%); }
  to { transform: translateY(0); }
}
@keyframes idkit-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}
@keyframes idkit-pulse {
  0%, 100% { opacity: 1; }
  50% { opacity: 0.5; }
}

/* Backdrop */
.idkit-backdrop {
  position: fixed;
  inset: 0;
  background: rgba(0, 0, 0, 0.5);
  backdrop-filter: blur(12px);
  -webkit-backdrop-filter: blur(12px);
  display: flex;
  align-items: center;
  justify-content: center;
  z-index: 2147483646;
  padding: 16px;
  animation: idkit-fade-in 0.2s ease-out;
}

/* Modal */
.idkit-modal {
  position: relative;
  width: 100%;
  max-width: 448px;
  min-height: 35rem;
  background: var(--idkit-bg);
  border-radius: 24px;
  box-shadow: 0 24px 60px rgba(15, 23, 42, 0.28), 0 0 0 1px rgba(0, 0, 0, 0.04);
  font-family: var(--idkit-font);
  color: var(--idkit-text);
  display: flex;
  flex-direction: column;
  overflow: hidden;
  animation: idkit-scale-in 0.25s ease-out;
}

/* Close button */
.idkit-close {
  position: absolute;
  top: 24px;
  right: 24px;
  display: flex;
  align-items: center;
  justify-content: center;
  width: 32px;
  height: 32px;
  border: 1.2px solid var(--idkit-border);
  border-radius: 50%;
  background: transparent;
  color: var(--idkit-text);
  cursor: pointer;
  padding: 0;
  z-index: 1;
  transition: background 0.15s ease;
}
.idkit-close:hover {
  background: var(--idkit-surface);
}
.idkit-close svg {
  width: 16px;
  height: 16px;
}

/* Content area */
.idkit-content {
  flex: 1;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  padding: 24px 24px 24px;
  text-align: center;
}

/* Footer */
.idkit-footer {
  border-top: 1px solid #F5F5F7;
  padding: 28px 32px;
  text-align: center;
}
:host(.dark) .idkit-footer {
  border-top-color: rgba(245, 245, 247, 0.15);
}
.idkit-footer a {
  color: var(--idkit-text-muted);
  font-size: 14px;
  text-decoration: none;
  transition: color 0.15s ease;
}
.idkit-footer a:hover {
  color: var(--idkit-text);
}

/* World ID State */
.idkit-worldid-icon {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 56px;
  height: 56px;
  border-radius: 50%;
  border: 1.2px solid var(--idkit-border);
  margin-bottom: 16px;
}
.idkit-worldid-icon svg {
  width: 32px;
  height: 32px;
  color: var(--idkit-text);
}

.idkit-heading {
  margin: 0;
  font-size: 24px;
  font-weight: 600;
  line-height: 1.2;
  color: var(--idkit-text);
}

.idkit-subtext {
  margin: 12px 0 0;
  font-size: 16px;
  line-height: 1.4;
  color: var(--idkit-text-muted);
}

/* QR Container */
.idkit-qr-container {
  position: relative;
  width: 100%;
  margin-top: 40px;
}

.idkit-qr-overlay {
  position: absolute;
  inset: 0;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  gap: 24px;
  z-index: 2;
}

.idkit-qr-blur {
  transition: filter 0.5s ease-in-out, opacity 0.5s ease-in-out;
}
.idkit-qr-blur.blurred {
  filter: blur(16px);
  opacity: 0.4;
}

.idkit-qr-wrapper {
  display: inline-flex;
  align-items: center;
  justify-content: center;
  border-radius: 16px;
  border: 1px solid var(--idkit-border-light);
  padding: 12px;
}

.idkit-qr-inner {
  color: var(--idkit-text);
}

/* QR Code SVG colors for finder patterns */
.idkit-qr-inner .qr-finder-dark { color: #000000; }
.idkit-qr-inner .qr-finder-light { color: #ffffff; }
:host(.dark) .idkit-qr-inner .qr-finder-dark { color: #000000; }
:host(.dark) .idkit-qr-inner .qr-finder-light { color: #ffffff; }
.idkit-qr-inner .qr-dot { color: #000000; }
:host(.dark) .idkit-qr-inner .qr-dot { color: #ffffff; }

/* QR Placeholder */
.idkit-qr-placeholder {
  width: 200px;
  height: 200px;
  display: flex;
  align-items: center;
  justify-content: center;
}
.idkit-qr-placeholder svg {
  width: 200px;
  height: 200px;
  animation: idkit-pulse 2s cubic-bezier(0.4, 0, 0.6, 1) infinite;
}

/* Copy toast \u2014 absolutely positioned so it never shifts layout */
.idkit-copy-toast {
  position: absolute;
  bottom: 100%;
  left: 50%;
  transform: translateX(-50%);
  margin-bottom: 8px;
  opacity: 0;
  transition: opacity 0.25s ease-in-out;
  pointer-events: none;
  white-space: nowrap;
}
.idkit-copy-toast.visible {
  opacity: 1;
  transition: opacity 0.2s ease-in-out 0.05s;
}
.idkit-copy-toast > span {
  display: inline-block;
  border-radius: 8px;
  border: 1px solid var(--idkit-border-light);
  padding: 4px 8px;
  font-size: 14px;
  color: var(--idkit-text-secondary);
}

/* Simulator callout (staging only) */
.idkit-simulator-callout {
  margin: 12px 0 0;
  font-size: 14px;
  color: var(--idkit-text-secondary);
  text-align: center;
}
.idkit-simulator-callout a {
  color: var(--idkit-text-muted);
  text-decoration: underline;
  text-underline-offset: 2px;
}
.idkit-simulator-callout a:hover {
  color: var(--idkit-text);
}

/* Mobile deep-link button */
.idkit-deeplink-btn {
  display: flex;
  width: 100%;
  align-items: center;
  gap: 8px;
  border-radius: 16px;
  border: 1px solid transparent;
  background: var(--idkit-btn-bg);
  color: var(--idkit-btn-text);
  padding: 16px;
  font-family: var(--idkit-font);
  font-size: 16px;
  font-weight: 500;
  text-decoration: none;
  cursor: pointer;
  box-shadow: 0 1px 3px rgba(0, 0, 0, 0.1);
}
.idkit-deeplink-btn svg {
  width: 20px;
  height: 20px;
  flex-shrink: 0;
}
.idkit-deeplink-btn span {
  flex: 1;
  text-align: center;
}

/* Loading spinner */
.idkit-spinner {
  animation: idkit-spin 1s linear infinite;
}
.idkit-spinner svg {
  width: 24px;
  height: 24px;
}

.idkit-connecting-text {
  text-align: center;
}
.idkit-connecting-text p:first-child {
  font-weight: 500;
  color: var(--idkit-text-muted);
  margin: 0;
}
.idkit-connecting-text p:last-child {
  font-size: 14px;
  font-weight: 300;
  color: var(--idkit-text-muted);
  margin: 4px 0 0;
}

/* Success State */
.idkit-success-icon {
  display: flex;
  align-items: center;
  justify-content: center;
  margin-bottom: 24px;
}
.idkit-success-icon svg {
  width: 96px;
  height: 96px;
}

/* Error State */
.idkit-error-icon {
  display: flex;
  align-items: center;
  justify-content: center;
  margin-bottom: 24px;
}
.idkit-error-icon svg {
  width: 96px;
  height: 96px;
}

.idkit-error-title {
  margin: 0;
  font-size: 24px;
  font-weight: 600;
  color: var(--idkit-text);
}

.idkit-error-message {
  margin: 8px auto 0;
  max-width: 224px;
  font-size: 16px;
  color: var(--idkit-text-muted);
  line-height: 1.4;
}

/* Retry button */
.idkit-retry-btn {
  display: inline-flex;
  align-items: center;
  border-radius: 9999px;
  border: 1.2px solid var(--idkit-border);
  background: transparent;
  padding: 12px 32px;
  font-family: var(--idkit-font);
  font-size: 16px;
  font-weight: 600;
  color: var(--idkit-text);
  cursor: pointer;
  transition: box-shadow 0.3s ease;
  margin-top: 32px;
}
.idkit-retry-btn:hover {
  box-shadow: 0 4px 12px rgba(0, 0, 0, 0.1);
}

/* Success subtext: old SDK used text-lg (18px) */
.idkit-success-icon ~ .idkit-subtext {
  font-size: 18px;
  line-height: 1.56;
}

/* Responsive: mobile full-screen bottom-anchored */
@media (max-width: 1024px) {
  .idkit-backdrop {
    align-items: flex-end;
    padding: 0;
  }
  .idkit-modal {
    max-width: 100%;
    min-height: auto;
    max-height: 95vh;
    border-radius: 24px 24px 0 0;
    animation: idkit-slide-up 0.3s ease-out;
  }

  /* Hide desktop QR on mobile */
  .idkit-desktop-only {
    display: none;
  }
  .idkit-mobile-only {
    display: block;
  }
}

@media (min-width: 1025px) {
  .idkit-mobile-only {
    display: none;
  }
  .idkit-desktop-only {
    display: block;
    position: relative;
  }
  .idkit-subtext {
    margin-top: 8px;
  }
}
`;
function ShadowHost({ children }) {
  const [root, setRoot] = react.useState(null);
  react.useEffect(() => {
    if (typeof document === "undefined") {
      return;
    }
    const host = document.createElement("div");
    host.setAttribute("data-idkit-shadow-host", "true");
    document.body.appendChild(host);
    const shadowRoot = host.attachShadow({ mode: "open" });
    setRoot(shadowRoot);
    return () => {
      host.remove();
      setRoot(null);
    };
  }, []);
  if (!root) {
    return null;
  }
  return reactDom.createPortal(children, root);
}
function XMarkIcon(props) {
  return /* @__PURE__ */ jsxRuntime.jsx(
    "svg",
    {
      xmlns: "http://www.w3.org/2000/svg",
      fill: "none",
      viewBox: "0 0 24 24",
      ...props,
      children: /* @__PURE__ */ jsxRuntime.jsx(
        "path",
        {
          strokeWidth: "1.5",
          stroke: "currentColor",
          strokeLinecap: "round",
          strokeLinejoin: "round",
          d: "m16.243 7.758-8.485 8.485m8.485 0L7.758 7.758"
        }
      )
    }
  );
}

// src/lang/translations/en.ts
var en = {
  "All set!": "All set!",
  "Your World ID is now connected": "Your World ID is now connected",
  "Something went wrong": "Something went wrong",
  "Request cancelled": "Request cancelled",
  "You've cancelled the request in World App.": "You've cancelled the request in World App.",
  "Connection lost": "Connection lost",
  "Please check your connection and try again.": "Please check your connection and try again.",
  "We couldn't complete your request. Please try again.": "We couldn't complete your request. Please try again.",
  "Try Again": "Try Again",
  "Open World App": "Open World App",
  "QR Code copied": "QR Code copied",
  "Connect your World ID": "Connect your World ID",
  "Use phone camera to scan the QR code": "Use phone camera to scan the QR code",
  "Connecting...": "Connecting...",
  "Please continue in app": "Please continue in app",
  "You will be redirected to the app, please return to this page once you're done": "You will be redirected to the app, please return to this page once you're done",
  "Terms & Privacy": "Terms & Privacy"
};

// src/lang/translations/es.ts
var es = {
  "All set!": "\xA1Todo listo!",
  "Your World ID is now connected": "Tu World ID ahora est\xE1 conectado",
  "Something went wrong": "Algo sali\xF3 mal",
  "Request cancelled": "Solicitud cancelada",
  "You've cancelled the request in World App.": "Has cancelado la solicitud en World App.",
  "Connection lost": "Conexion perdida",
  "Please check your connection and try again.": "Por favor verifica tu conexion e intenta de nuevo.",
  "We couldn't complete your request. Please try again.": "No pudimos completar tu solicitud. Por favor intenta de nuevo.",
  "Try Again": "Intentar de nuevo",
  "Open World App": "Abrir World App",
  "QR Code copied": "C\xF3digo QR copiado",
  "Connect your World ID": "Conecta tu World ID",
  "Use phone camera to scan the QR code": "Usa la c\xE1mara del tel\xE9fono para escanear el c\xF3digo QR",
  "Connecting...": "Conectando...",
  "Please continue in app": "Por favor contin\xFAa en la aplicaci\xF3n",
  "You will be redirected to the app, please return to this page once you're done": "Ser\xE1s redirigido a la aplicaci\xF3n, por favor regresa a esta p\xE1gina una vez que hayas terminado",
  "Terms & Privacy": "T\xE9rminos y privacidad"
};

// src/lang/translations/th.ts
var th = {
  "All set!": "\u0E40\u0E23\u0E35\u0E22\u0E1A\u0E23\u0E49\u0E2D\u0E22!",
  "Your World ID is now connected": "World ID \u0E02\u0E2D\u0E07\u0E04\u0E38\u0E13\u0E40\u0E0A\u0E37\u0E48\u0E2D\u0E21\u0E15\u0E48\u0E2D\u0E41\u0E25\u0E49\u0E27",
  "Something went wrong": "\u0E40\u0E01\u0E34\u0E14\u0E02\u0E49\u0E2D\u0E1C\u0E34\u0E14\u0E1E\u0E25\u0E32\u0E14",
  "Request cancelled": "\u0E04\u0E33\u0E02\u0E2D\u0E16\u0E39\u0E01\u0E22\u0E01\u0E40\u0E25\u0E34\u0E01",
  "You've cancelled the request in World App.": "\u0E04\u0E38\u0E13\u0E44\u0E14\u0E49\u0E22\u0E01\u0E40\u0E25\u0E34\u0E01\u0E04\u0E33\u0E02\u0E2D\u0E43\u0E19 World App",
  "Connection lost": "\u0E01\u0E32\u0E23\u0E40\u0E0A\u0E37\u0E48\u0E2D\u0E21\u0E15\u0E48\u0E2D\u0E02\u0E32\u0E14\u0E2B\u0E32\u0E22",
  "Please check your connection and try again.": "\u0E01\u0E23\u0E38\u0E13\u0E32\u0E15\u0E23\u0E27\u0E08\u0E2A\u0E2D\u0E1A\u0E01\u0E32\u0E23\u0E40\u0E0A\u0E37\u0E48\u0E2D\u0E21\u0E15\u0E48\u0E2D\u0E41\u0E25\u0E49\u0E27\u0E25\u0E2D\u0E07\u0E2D\u0E35\u0E01\u0E04\u0E23\u0E31\u0E49\u0E07",
  "We couldn't complete your request. Please try again.": "\u0E40\u0E23\u0E32\u0E44\u0E21\u0E48\u0E2A\u0E32\u0E21\u0E32\u0E23\u0E16\u0E14\u0E33\u0E40\u0E19\u0E34\u0E19\u0E01\u0E32\u0E23\u0E15\u0E32\u0E21\u0E04\u0E33\u0E02\u0E2D\u0E02\u0E2D\u0E07\u0E04\u0E38\u0E13\u0E44\u0E14\u0E49 \u0E01\u0E23\u0E38\u0E13\u0E32\u0E25\u0E2D\u0E07\u0E2D\u0E35\u0E01\u0E04\u0E23\u0E31\u0E49\u0E07",
  "Try Again": "\u0E25\u0E2D\u0E07\u0E2D\u0E35\u0E01\u0E04\u0E23\u0E31\u0E49\u0E07",
  "Open World App": "\u0E40\u0E1B\u0E34\u0E14 World App",
  "QR Code copied": "\u0E04\u0E31\u0E14\u0E25\u0E2D\u0E01 QR Code \u0E41\u0E25\u0E49\u0E27",
  "Connect your World ID": "\u0E40\u0E0A\u0E37\u0E48\u0E2D\u0E21\u0E15\u0E48\u0E2D World ID \u0E02\u0E2D\u0E07\u0E04\u0E38\u0E13",
  "Use phone camera to scan the QR code": "\u0E43\u0E0A\u0E49\u0E01\u0E25\u0E49\u0E2D\u0E07\u0E42\u0E17\u0E23\u0E28\u0E31\u0E1E\u0E17\u0E4C\u0E40\u0E1E\u0E37\u0E48\u0E2D\u0E2A\u0E41\u0E01\u0E19 QR code",
  "Connecting...": "\u0E01\u0E33\u0E25\u0E31\u0E07\u0E40\u0E0A\u0E37\u0E48\u0E2D\u0E21\u0E15\u0E48\u0E2D...",
  "Please continue in app": "\u0E01\u0E23\u0E38\u0E13\u0E32\u0E14\u0E33\u0E40\u0E19\u0E34\u0E19\u0E01\u0E32\u0E23\u0E15\u0E48\u0E2D\u0E43\u0E19\u0E41\u0E2D\u0E1B",
  "You will be redirected to the app, please return to this page once you're done": "\u0E04\u0E38\u0E13\u0E08\u0E30\u0E16\u0E39\u0E01\u0E19\u0E33\u0E44\u0E1B\u0E22\u0E31\u0E07\u0E41\u0E2D\u0E1B \u0E01\u0E23\u0E38\u0E13\u0E32\u0E01\u0E25\u0E31\u0E1A\u0E21\u0E32\u0E17\u0E35\u0E48\u0E2B\u0E19\u0E49\u0E32\u0E19\u0E35\u0E49\u0E40\u0E21\u0E37\u0E48\u0E2D\u0E40\u0E2A\u0E23\u0E47\u0E08\u0E41\u0E25\u0E49\u0E27",
  "Terms & Privacy": "\u0E02\u0E49\u0E2D\u0E01\u0E33\u0E2B\u0E19\u0E14\u0E41\u0E25\u0E30\u0E04\u0E27\u0E32\u0E21\u0E40\u0E1B\u0E47\u0E19\u0E2A\u0E48\u0E27\u0E19\u0E15\u0E31\u0E27"
};

// src/lang/translations/index.ts
var translations = {
  en,
  es,
  th
};

// src/lang/localization.ts
var currentConfig = {};
var setLocalizationConfig = (config) => {
  currentConfig = config;
};
var getLocalizationConfig = () => currentConfig;
var detectBrowserLanguage = () => {
  if (typeof navigator === "undefined") return void 0;
  for (const lang of navigator.languages) {
    const [language] = lang.split("-");
    const normalizedLang = language.toLowerCase();
    if (normalizedLang in translations) {
      return normalizedLang;
    }
  }
  return void 0;
};
var getCurrentLanguage = () => {
  const config = getLocalizationConfig();
  if (config.language && config.language in translations) {
    return config.language;
  }
  const browserLang = detectBrowserLanguage();
  if (browserLang) {
    return browserLang;
  }
  return "en";
};
var getTranslations = () => {
  const currentLang = getCurrentLanguage();
  return translations[currentLang];
};

// src/lang/index.ts
var getLang = () => {
  return getTranslations();
};
var replaceParams = (str, params) => {
  if (!params) return str;
  let result = str;
  for (const [key, value] of Object.entries(params)) {
    result = result.replace(`:${key}`, value);
  }
  return result;
};
function __(str, ...args) {
  const [params] = args;
  if (typeof navigator === "undefined" && typeof window === "undefined") {
    return replaceParams(str, params);
  }
  const translated = getLang()?.[str] ?? str;
  return replaceParams(translated, params);
}
function ModalContent({
  onOpenChange,
  children
}) {
  return /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, { children: [
    /* @__PURE__ */ jsxRuntime.jsx("style", { children: WIDGET_STYLES }),
    /* @__PURE__ */ jsxRuntime.jsx(
      "div",
      {
        className: "idkit-backdrop",
        role: "presentation",
        onClick: () => onOpenChange(false),
        children: /* @__PURE__ */ jsxRuntime.jsxs(
          "section",
          {
            className: "idkit-modal",
            role: "dialog",
            "aria-modal": "true",
            onClick: (event) => event.stopPropagation(),
            children: [
              /* @__PURE__ */ jsxRuntime.jsx(
                "button",
                {
                  type: "button",
                  className: "idkit-close",
                  onClick: () => onOpenChange(false),
                  "aria-label": "Close",
                  children: /* @__PURE__ */ jsxRuntime.jsx(XMarkIcon, {})
                }
              ),
              /* @__PURE__ */ jsxRuntime.jsx("div", { className: "idkit-content", children }),
              /* @__PURE__ */ jsxRuntime.jsx("footer", { className: "idkit-footer", children: /* @__PURE__ */ jsxRuntime.jsx(
                "a",
                {
                  href: "https://developer.world.org/privacy-statement",
                  target: "_blank",
                  rel: "noopener noreferrer",
                  children: __("Terms & Privacy")
                }
              ) })
            ]
          }
        )
      }
    )
  ] });
}
function IDKitModal({
  open,
  onOpenChange,
  children
}) {
  react.useEffect(() => {
    if (!open || typeof document === "undefined") {
      return;
    }
    const handleEsc = (event) => {
      if (event.key === "Escape") {
        onOpenChange(false);
      }
    };
    window.addEventListener("keydown", handleEsc);
    return () => window.removeEventListener("keydown", handleEsc);
  }, [onOpenChange, open]);
  if (!open || typeof document === "undefined") {
    return null;
  }
  const content = /* @__PURE__ */ jsxRuntime.jsx(ModalContent, { onOpenChange, children });
  return /* @__PURE__ */ jsxRuntime.jsx(ShadowHost, { children: content });
}
function useMedia() {
  const getInitialState = () => {
    if (typeof window !== "undefined") {
      return window.matchMedia("(max-width: 1024px)").matches ? "mobile" : "desktop";
    }
    return "desktop";
  };
  const [media, setMedia] = react.useState(getInitialState);
  react.useEffect(() => {
    const mql = window.matchMedia("(max-width: 1024px)");
    const handleChange = (e) => setMedia(e.matches ? "mobile" : "desktop");
    handleChange(mql);
    mql.addEventListener("change", handleChange);
    return () => {
      mql.removeEventListener("change", handleChange);
    };
  }, []);
  return media;
}
function WorldcoinIcon(props) {
  return /* @__PURE__ */ jsxRuntime.jsx(
    "svg",
    {
      xmlns: "http://www.w3.org/2000/svg",
      width: "32",
      height: "32",
      viewBox: "0 0 32 32",
      fill: "none",
      ...props,
      children: /* @__PURE__ */ jsxRuntime.jsx(
        "path",
        {
          d: "M30.7367 9.77239C29.9301 7.86586 28.7772 6.15721 27.3084 4.68831C25.8397 3.21941 24.1275 2.06636 22.225 1.2596C20.2502 0.422405 18.1574 0 15.9962 0C13.8388 0 11.7422 0.422405 9.76742 1.2596C7.86112 2.06636 6.15268 3.21941 4.68395 4.68831C3.21522 6.15721 2.06231 7.86966 1.25565 9.77239C0.422354 11.7436 0 13.8404 0 15.9981C0 18.1558 0.422354 20.2526 1.25945 22.2276C2.06611 24.1341 3.21903 25.8428 4.68775 27.3117C6.15648 28.7806 7.86873 29.9336 9.77122 30.7404C11.746 31.5738 13.8388 32 16 32C18.1574 32 20.254 31.5776 22.2288 30.7404C24.1351 29.9336 25.8435 28.7806 27.3122 27.3117C28.781 25.8428 29.9339 24.1303 30.7405 22.2276C31.5738 20.2526 32 18.1596 32 15.9981C31.9962 13.8404 31.57 11.7436 30.7367 9.77239ZM10.6844 14.4949C11.3503 11.9377 13.679 10.0464 16.4452 10.0464H27.552C28.2673 11.4278 28.7239 12.9309 28.9027 14.4949H10.6844ZM28.9027 17.5012C28.7239 19.0653 28.2635 20.5684 27.552 21.9498H16.4452C13.6828 21.9498 11.3541 20.0585 10.6844 17.5012H28.9027ZM6.81094 6.81175C9.26516 4.35724 12.526 3.0063 15.9962 3.0063C19.4663 3.0063 22.7272 4.35724 25.1815 6.81175C25.2576 6.88786 25.3298 6.96397 25.4021 7.04008H16.4452C14.0518 7.04008 11.8031 7.97241 10.1099 9.66583C8.77812 10.9977 7.91819 12.6759 7.60999 14.4988H3.09346C3.42449 11.5952 4.71439 8.90855 6.81094 6.81175ZM15.9962 28.9937C12.526 28.9937 9.26516 27.6428 6.81094 25.1883C4.71439 23.0915 3.42449 20.4048 3.09346 17.5051H7.60999C7.91439 19.3279 8.77812 21.0061 10.1099 22.338C11.8031 24.0314 14.0518 24.9637 16.4452 24.9637H25.4059C25.3337 25.0398 25.2576 25.1159 25.1853 25.1921C22.731 27.639 19.4663 28.9937 15.9962 28.9937Z",
          fill: "currentColor"
        }
      )
    }
  );
}
function LoadingIcon(props) {
  return /* @__PURE__ */ jsxRuntime.jsxs(
    "svg",
    {
      xmlns: "http://www.w3.org/2000/svg",
      fill: "none",
      viewBox: "0 0 24 24",
      ...props,
      children: [
        /* @__PURE__ */ jsxRuntime.jsx(
          "circle",
          {
            cx: "12",
            cy: "12",
            r: "10.75",
            stroke: "#191C20",
            strokeOpacity: ".16",
            strokeWidth: "2.5"
          }
        ),
        /* @__PURE__ */ jsxRuntime.jsx(
          "path",
          {
            fill: "#191C20",
            d: "M17.28 2.633c.338-.6.127-1.368-.505-1.642A12 12 0 0 0 7.459.892c-.638.261-.864 1.024-.539 1.632.326.607 1.08.827 1.725.584a9.504 9.504 0 0 1 6.897.073c.64.257 1.399.053 1.737-.548Z"
          }
        )
      ]
    }
  );
}
var generateMatrix = (data) => {
  const arr = QRCodeUtil__default.default.create(data, { errorCorrectionLevel: "M" }).modules.data;
  const sqrt = Math.sqrt(arr.length);
  return arr.reduce(
    (rows, key, index) => {
      if (index % sqrt === 0) rows.push([key]);
      else rows[rows.length - 1].push(key);
      return rows;
    },
    []
  );
};
function QRCodeInner({ data, size = 200 }) {
  const dots = react.useMemo(() => {
    const elements = [];
    const matrix = generateMatrix(data);
    const cellSize = size / matrix.length;
    const qrList = [
      { x: 0, y: 0 },
      { x: 1, y: 0 },
      { x: 0, y: 1 }
    ];
    qrList.forEach(({ x, y }) => {
      const x1 = (matrix.length - 7) * cellSize * x;
      const y1 = (matrix.length - 7) * cellSize * y;
      for (let i = 0; i < 3; i++) {
        elements.push(
          /* @__PURE__ */ jsxRuntime.jsx(
            "rect",
            {
              fill: "currentColor",
              x: x1 + cellSize * i,
              y: y1 + cellSize * i,
              width: cellSize * (7 - i * 2),
              height: cellSize * (7 - i * 2),
              rx: (i - 2) * -5,
              ry: (i - 2) * -5,
              className: i % 3 === 0 ? "qr-finder-dark" : i % 3 === 1 ? "qr-finder-light" : "qr-finder-dark"
            },
            `${i}-${x}-${y}`
          )
        );
      }
    });
    matrix.forEach((row, i) => {
      row.forEach((_, j) => {
        if (!matrix[i][j]) return;
        if (i < 7 && j < 7 || i > matrix.length - 8 && j < 7 || i < 7 && j > matrix.length - 8)
          return;
        elements.push(
          /* @__PURE__ */ jsxRuntime.jsx(
            "circle",
            {
              fill: "currentColor",
              r: cellSize / 2.2,
              cx: i * cellSize + cellSize / 2,
              cy: j * cellSize + cellSize / 2,
              className: "qr-dot"
            },
            `circle-${i}-${j}`
          )
        );
      });
    });
    return elements;
  }, [size, data]);
  return /* @__PURE__ */ jsxRuntime.jsx("svg", { height: size, width: size, "data-test-id": "qr-code", children: dots });
}
var QRCode = react.memo(QRCodeInner);
function QRPlaceholderIcon(props) {
  return /* @__PURE__ */ jsxRuntime.jsxs(
    "svg",
    {
      xmlns: "http://www.w3.org/2000/svg",
      fill: "none",
      viewBox: "0 0 200 200",
      ...props,
      children: [
        /* @__PURE__ */ jsxRuntime.jsx(
          "path",
          {
            fill: "#EBECEF",
            fillRule: "evenodd",
            d: "M12.1 0C5.417 0 0 5.417 0 12.1v18.505c0 6.682 5.417 12.1 12.1 12.1h18.505c6.682 0 12.1-5.418 12.1-12.1V12.1c0-6.683-5.418-12.1-12.1-12.1H12.1Zm18.505 11.388H12.1a.712.712 0 0 0-.712.712v18.505c0 .393.319.712.712.712h18.505a.712.712 0 0 0 .712-.712V12.1a.712.712 0 0 0-.712-.712Z",
            clipRule: "evenodd"
          }
        ),
        /* @__PURE__ */ jsxRuntime.jsx(
          "path",
          {
            fill: "#EBECEF",
            d: "M197.026 200c.789 0 1.545-.309 2.103-.86.558-.55.871-1.297.871-2.076v-17.616c0-.778-.313-1.525-.871-2.076a2.996 2.996 0 0 0-2.103-.86h-5.948c-.789 0-1.545.31-2.103.86a2.918 2.918 0 0 0-.871 2.076v8.808h-11.897v-11.744h-11.896v-23.487h11.896v8.808c0 .778.314 1.525.872 2.076.557.55 1.314.86 2.102.86h5.949c.788 0 1.545-.31 2.103-.86a2.922 2.922 0 0 0 .871-2.076v-8.808h8.922c.789 0 1.545-.309 2.103-.86a2.916 2.916 0 0 0 .871-2.076v-5.872c0-.779-.313-1.525-.871-2.076a2.992 2.992 0 0 0-2.103-.86h-29.741c-.789 0-1.545.309-2.103.86a2.916 2.916 0 0 0-.871 2.076v8.808h-23.792v-11.744h8.922c.789 0 1.545-.309 2.103-.86.558-.55.871-1.297.871-2.076v-5.872c0-.778-.313-1.525-.871-2.076a2.996 2.996 0 0 0-2.103-.86h-5.948c-.789 0-1.546.31-2.103.86a2.918 2.918 0 0 0-.871 2.076v8.808H119.7c-.789 0-1.545.309-2.103.86a2.916 2.916 0 0 0-.871 2.076v5.872c0 .779.313 1.525.871 2.076.558.551 1.314.86 2.103.86h20.819v8.808c0 .778.313 1.525.871 2.076.557.55 1.314.86 2.103.86h8.922v8.808c0 .778.313 1.525.871 2.076.558.55 1.314.859 2.103.859h8.922v11.744h-20.818c-.789 0-1.546.31-2.103.86a2.916 2.916 0 0 0-.871 2.076v5.872c0 .779.313 1.526.871 2.076a2.99 2.99 0 0 0 2.103.86h17.844c.789 0 1.545-.309 2.103-.86.558-.55.871-1.297.871-2.076v-8.808h11.896v8.808c0 .779.314 1.526.872 2.076a2.99 2.99 0 0 0 2.102.86h17.845Z"
          }
        ),
        /* @__PURE__ */ jsxRuntime.jsx(
          "path",
          {
            fill: "#EBECEF",
            fillRule: "evenodd",
            d: "M157.295 12.1c0-6.683 5.418-12.1 12.1-12.1H187.9c6.683 0 12.1 5.417 12.1 12.1v18.505c0 6.682-5.417 12.1-12.1 12.1h-18.505c-6.682 0-12.1-5.418-12.1-12.1V12.1Zm12.1-.712H187.9c.393 0 .712.319.712.712v18.505a.712.712 0 0 1-.712.712h-18.505a.712.712 0 0 1-.712-.712V12.1c0-.393.319-.712.712-.712ZM12.1 157.295c-6.683 0-12.1 5.418-12.1 12.1V187.9c0 6.683 5.417 12.1 12.1 12.1h18.505c6.682 0 12.1-5.417 12.1-12.1v-18.505c0-6.682-5.418-12.1-12.1-12.1H12.1Zm19.217 12.1a.712.712 0 0 0-.712-.712H12.1a.712.712 0 0 0-.712.712V187.9c0 .393.319.712.712.712h18.505a.712.712 0 0 0 .712-.712v-18.505Z",
            clipRule: "evenodd"
          }
        ),
        /* @__PURE__ */ jsxRuntime.jsx(
          "path",
          {
            fill: "#EBECEF",
            d: "M6.05 89.68A6.05 6.05 0 0 0 0 95.73v9.252a6.05 6.05 0 0 0 6.05 6.05h9.253a6.05 6.05 0 0 0 6.05-6.05V95.73c0-.678-.112-1.33-.318-1.94.445.105.908.16 1.385.16h27.758a6.05 6.05 0 0 0 6.05-6.05v-9.252a6.05 6.05 0 0 0-6.05-6.05H22.42a6.05 6.05 0 0 0-6.05 6.05V87.9c0 .678.112 1.33.317 1.939a6.065 6.065 0 0 0-1.385-.16H6.05Zm102.135-40.926a6.05 6.05 0 0 1 6.05-6.05h9.253a6.05 6.05 0 0 1 6.049 6.05v9.253a6.05 6.05 0 0 1-6.049 6.05h-9.253a6.05 6.05 0 0 1-6.05-6.05v-9.253ZM67.616 184.698a6.05 6.05 0 0 1 6.05-6.05h9.252c.678 0 1.33.111 1.939.317a6.064 6.064 0 0 1-.16-1.385v-9.253a6.05 6.05 0 0 1 6.05-6.049H100a6.05 6.05 0 0 1 6.05 6.049v9.253a6.05 6.05 0 0 1-6.05 6.05h-9.253c-.678 0-1.33-.112-1.938-.317.104.444.159.908.159 1.385v9.252a6.05 6.05 0 0 1-6.05 6.05h-9.253a6.05 6.05 0 0 1-6.05-6.05v-9.252Zm78.291-120.285a6.05 6.05 0 0 1 6.05-6.05h41.993a6.05 6.05 0 0 1 6.05 6.05v9.252a6.05 6.05 0 0 1-6.05 6.05h-41.993a6.05 6.05 0 0 1-6.05-6.05v-9.252ZM95.018 0a6.05 6.05 0 0 0-6.05 6.05v17.082a6.05 6.05 0 0 0 6.05 6.05h9.252a6.05 6.05 0 0 0 6.05-6.05V6.05A6.05 6.05 0 0 0 104.27 0h-9.252Z"
          }
        ),
        /* @__PURE__ */ jsxRuntime.jsx(
          "path",
          {
            fill: "url(#idkit_qr_placeholder_gradient)",
            fillRule: "evenodd",
            d: "M12.1 0C5.417 0 0 5.417 0 12.1v18.505c0 6.682 5.417 12.1 12.1 12.1h18.505c6.682 0 12.1-5.418 12.1-12.1V12.1c0-6.683-5.418-12.1-12.1-12.1H12.1Zm18.505 11.388H12.1a.712.712 0 0 0-.712.712v18.505c0 .393.319.712.712.712h18.505a.712.712 0 0 0 .712-.712V12.1a.712.712 0 0 0-.712-.712Z",
            clipRule: "evenodd"
          }
        ),
        /* @__PURE__ */ jsxRuntime.jsx(
          "path",
          {
            fill: "url(#idkit_qr_placeholder_gradient)",
            d: "M197.026 200c.789 0 1.545-.309 2.103-.86.558-.55.871-1.297.871-2.076v-17.616c0-.778-.313-1.525-.871-2.076a2.996 2.996 0 0 0-2.103-.86h-5.948c-.789 0-1.545.31-2.103.86a2.918 2.918 0 0 0-.871 2.076v8.808h-11.897v-11.744h-11.896v-23.487h11.896v8.808c0 .778.314 1.525.872 2.076.557.55 1.314.86 2.102.86h5.949c.788 0 1.545-.31 2.103-.86a2.922 2.922 0 0 0 .871-2.076v-8.808h8.922c.789 0 1.545-.309 2.103-.86a2.916 2.916 0 0 0 .871-2.076v-5.872c0-.779-.313-1.525-.871-2.076a2.992 2.992 0 0 0-2.103-.86h-29.741c-.789 0-1.545.309-2.103.86a2.916 2.916 0 0 0-.871 2.076v8.808h-23.792v-11.744h8.922c.789 0 1.545-.309 2.103-.86.558-.55.871-1.297.871-2.076v-5.872c0-.778-.313-1.525-.871-2.076a2.996 2.996 0 0 0-2.103-.86h-5.948c-.789 0-1.546.31-2.103.86a2.918 2.918 0 0 0-.871 2.076v8.808H119.7c-.789 0-1.545.309-2.103.86a2.916 2.916 0 0 0-.871 2.076v5.872c0 .779.313 1.525.871 2.076.558.551 1.314.86 2.103.86h20.819v8.808c0 .778.313 1.525.871 2.076.557.55 1.314.86 2.103.86h8.922v8.808c0 .778.313 1.525.871 2.076.558.55 1.314.859 2.103.859h8.922v11.744h-20.818c-.789 0-1.546.31-2.103.86a2.916 2.916 0 0 0-.871 2.076v5.872c0 .779.313 1.526.871 2.076a2.99 2.99 0 0 0 2.103.86h17.844c.789 0 1.545-.309 2.103-.86.558-.55.871-1.297.871-2.076v-8.808h11.896v8.808c0 .779.314 1.526.872 2.076a2.99 2.99 0 0 0 2.102.86h17.845Z"
          }
        ),
        /* @__PURE__ */ jsxRuntime.jsx(
          "path",
          {
            fill: "url(#idkit_qr_placeholder_gradient)",
            fillRule: "evenodd",
            d: "M157.295 12.1c0-6.683 5.418-12.1 12.1-12.1H187.9c6.683 0 12.1 5.417 12.1 12.1v18.505c0 6.682-5.417 12.1-12.1 12.1h-18.505c-6.682 0-12.1-5.418-12.1-12.1V12.1Zm12.1-.712H187.9c.393 0 .712.319.712.712v18.505a.712.712 0 0 1-.712.712h-18.505a.712.712 0 0 1-.712-.712V12.1c0-.393.319-.712.712-.712Z",
            clipRule: "evenodd"
          }
        ),
        /* @__PURE__ */ jsxRuntime.jsx(
          "path",
          {
            fill: "url(#idkit_qr_placeholder_gradient)",
            fillRule: "evenodd",
            d: "M12.1 157.295c-6.683 0-12.1 5.418-12.1 12.1V187.9c0 6.683 5.417 12.1 12.1 12.1h18.505c6.682 0 12.1-5.417 12.1-12.1v-18.505c0-6.682-5.418-12.1-12.1-12.1H12.1Zm19.217 12.1a.712.712 0 0 0-.712-.712H12.1a.712.712 0 0 0-.712.712V187.9c0 .393.319.712.712.712h18.505a.712.712 0 0 0 .712-.712v-18.505Z",
            clipRule: "evenodd"
          }
        ),
        /* @__PURE__ */ jsxRuntime.jsx(
          "path",
          {
            fill: "url(#idkit_qr_placeholder_gradient)",
            d: "M6.05 89.68A6.05 6.05 0 0 0 0 95.73v9.252a6.05 6.05 0 0 0 6.05 6.05h9.253a6.05 6.05 0 0 0 6.05-6.05V95.73c0-.678-.112-1.33-.318-1.94.445.105.908.16 1.385.16h27.758a6.05 6.05 0 0 0 6.05-6.05v-9.252a6.05 6.05 0 0 0-6.05-6.05H22.42a6.05 6.05 0 0 0-6.05 6.05V87.9c0 .678.112 1.33.317 1.939a6.065 6.065 0 0 0-1.385-.16H6.05Z"
          }
        ),
        /* @__PURE__ */ jsxRuntime.jsx(
          "path",
          {
            fill: "url(#idkit_qr_placeholder_gradient)",
            d: "M108.185 48.754a6.05 6.05 0 0 1 6.05-6.05h9.253a6.05 6.05 0 0 1 6.049 6.05v9.253a6.05 6.05 0 0 1-6.049 6.05h-9.253a6.05 6.05 0 0 1-6.05-6.05v-9.253Z"
          }
        ),
        /* @__PURE__ */ jsxRuntime.jsx(
          "path",
          {
            fill: "url(#idkit_qr_placeholder_gradient)",
            d: "M67.616 184.698a6.05 6.05 0 0 1 6.05-6.05h9.252c.678 0 1.33.111 1.939.317a6.064 6.064 0 0 1-.16-1.385v-9.253a6.05 6.05 0 0 1 6.05-6.049H100a6.05 6.05 0 0 1 6.05 6.049v9.253a6.05 6.05 0 0 1-6.05 6.05h-9.253c-.678 0-1.33-.112-1.938-.317.104.444.159.908.159 1.385v9.252a6.05 6.05 0 0 1-6.05 6.05h-9.253a6.05 6.05 0 0 1-6.05-6.05v-9.252Z"
          }
        ),
        /* @__PURE__ */ jsxRuntime.jsx(
          "path",
          {
            fill: "url(#idkit_qr_placeholder_gradient)",
            d: "M145.907 64.413a6.05 6.05 0 0 1 6.05-6.05h41.993a6.05 6.05 0 0 1 6.05 6.05v9.252a6.05 6.05 0 0 1-6.05 6.05h-41.993a6.05 6.05 0 0 1-6.05-6.05v-9.252Z"
          }
        ),
        /* @__PURE__ */ jsxRuntime.jsx(
          "path",
          {
            fill: "url(#idkit_qr_placeholder_gradient)",
            d: "M95.018 0a6.05 6.05 0 0 0-6.05 6.05v17.082a6.05 6.05 0 0 0 6.05 6.05h9.252a6.05 6.05 0 0 0 6.05-6.05V6.05A6.05 6.05 0 0 0 104.27 0h-9.252Z"
          }
        ),
        /* @__PURE__ */ jsxRuntime.jsx("defs", { children: /* @__PURE__ */ jsxRuntime.jsxs(
          "linearGradient",
          {
            id: "idkit_qr_placeholder_gradient",
            x1: "0",
            x2: "200",
            y1: "0",
            y2: "200",
            gradientUnits: "userSpaceOnUse",
            children: [
              /* @__PURE__ */ jsxRuntime.jsx("stop", { offset: ".37", stopColor: "#fff", stopOpacity: "0" }),
              /* @__PURE__ */ jsxRuntime.jsx("stop", { offset: ".5", stopColor: "#fff", stopOpacity: ".85" }),
              /* @__PURE__ */ jsxRuntime.jsx("stop", { offset: ".63", stopColor: "#fff", stopOpacity: "0" })
            ]
          }
        ) })
      ]
    }
  );
}
function QRState({
  qrData,
  showSimulatorCallout
}) {
  const media = useMedia();
  const [copiedLink, setCopiedLink] = react.useState(false);
  const copyLink = react.useCallback(() => {
    if (!qrData) return;
    void navigator.clipboard.writeText(qrData);
    setCopiedLink(true);
    setTimeout(() => setCopiedLink(false), 2e3);
  }, [qrData]);
  return /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, { children: [
    /* @__PURE__ */ jsxRuntime.jsx("div", { className: "idkit-mobile-only", children: /* @__PURE__ */ jsxRuntime.jsxs("a", { href: qrData ?? void 0, className: "idkit-deeplink-btn", children: [
      /* @__PURE__ */ jsxRuntime.jsx(WorldcoinIcon, {}),
      /* @__PURE__ */ jsxRuntime.jsx("span", { children: __("Open World App") })
    ] }) }),
    /* @__PURE__ */ jsxRuntime.jsxs("div", { className: "idkit-desktop-only", children: [
      /* @__PURE__ */ jsxRuntime.jsx(
        "div",
        {
          className: `idkit-copy-toast ${copiedLink ? "visible" : "hidden"}`,
          style: {
            textAlign: "center",
            fontSize: "14px",
            color: "var(--idkit-text-secondary)"
          },
          children: /* @__PURE__ */ jsxRuntime.jsx("span", { children: __("QR Code copied") })
        }
      ),
      /* @__PURE__ */ jsxRuntime.jsx("div", { className: "idkit-qr-wrapper", children: /* @__PURE__ */ jsxRuntime.jsx("div", { className: "idkit-qr-inner", children: qrData ? /* @__PURE__ */ jsxRuntime.jsx(
        "div",
        {
          onClick: copyLink,
          onKeyDown: (e) => {
            if (e.key === "Enter") copyLink();
          },
          role: "button",
          tabIndex: 0,
          style: { cursor: "pointer" },
          children: /* @__PURE__ */ jsxRuntime.jsx(QRCode, { data: qrData, size: media === "mobile" ? 160 : 200 })
        }
      ) : /* @__PURE__ */ jsxRuntime.jsx("div", { className: "idkit-qr-placeholder", children: /* @__PURE__ */ jsxRuntime.jsx(QRPlaceholderIcon, {}) }) }) }),
      showSimulatorCallout && qrData && /* @__PURE__ */ jsxRuntime.jsxs("p", { className: "idkit-simulator-callout", children: [
        "Testing in staging?",
        " ",
        /* @__PURE__ */ jsxRuntime.jsx(
          "a",
          {
            href: `https://simulator.worldcoin.org?connect_url=${encodeURIComponent(qrData)}`,
            target: "_blank",
            rel: "noopener noreferrer",
            children: "Use the simulator"
          }
        )
      ] })
    ] })
  ] });
}
function WorldIDState({
  connectorURI,
  isAwaitingUserConfirmation,
  showSimulatorCallout
}) {
  const media = useMedia();
  return /* @__PURE__ */ jsxRuntime.jsxs(
    "div",
    {
      style: {
        display: "flex",
        flexDirection: "column",
        alignItems: "center",
        textAlign: "center"
      },
      children: [
        /* @__PURE__ */ jsxRuntime.jsx("div", { className: "idkit-worldid-icon", children: /* @__PURE__ */ jsxRuntime.jsx(WorldcoinIcon, {}) }),
        /* @__PURE__ */ jsxRuntime.jsx("h2", { className: "idkit-heading", children: __("Connect your World ID") }),
        /* @__PURE__ */ jsxRuntime.jsx("p", { className: "idkit-subtext", children: media === "mobile" ? __(
          "You will be redirected to the app, please return to this page once you're done"
        ) : __("Use phone camera to scan the QR code") }),
        /* @__PURE__ */ jsxRuntime.jsxs("div", { className: "idkit-qr-container", children: [
          isAwaitingUserConfirmation && /* @__PURE__ */ jsxRuntime.jsxs("div", { className: "idkit-qr-overlay", children: [
            /* @__PURE__ */ jsxRuntime.jsx("div", { className: "idkit-spinner", children: /* @__PURE__ */ jsxRuntime.jsx(LoadingIcon, {}) }),
            /* @__PURE__ */ jsxRuntime.jsxs("div", { className: "idkit-connecting-text", children: [
              /* @__PURE__ */ jsxRuntime.jsx("p", { children: __("Connecting...") }),
              /* @__PURE__ */ jsxRuntime.jsx("p", { children: __("Please continue in app") })
            ] })
          ] }),
          /* @__PURE__ */ jsxRuntime.jsx(
            "div",
            {
              className: `idkit-qr-blur ${isAwaitingUserConfirmation ? "blurred" : ""}`,
              children: /* @__PURE__ */ jsxRuntime.jsx("div", { style: { display: "flex", justifyContent: "center" }, children: /* @__PURE__ */ jsxRuntime.jsx(
                QRState,
                {
                  qrData: connectorURI,
                  showSimulatorCallout
                }
              ) })
            }
          )
        ] })
      ]
    }
  );
}
function CheckIcon(props) {
  return /* @__PURE__ */ jsxRuntime.jsxs(
    "svg",
    {
      width: "88",
      height: "88",
      viewBox: "0 0 88 88",
      fill: "none",
      xmlns: "http://www.w3.org/2000/svg",
      ...props,
      children: [
        /* @__PURE__ */ jsxRuntime.jsx("rect", { width: "88", height: "88", rx: "44", fill: "#00C230" }),
        /* @__PURE__ */ jsxRuntime.jsx(
          "rect",
          {
            opacity: "0.2",
            width: "88",
            height: "88",
            rx: "44",
            fill: "url(#idkit_check_radial)"
          }
        ),
        /* @__PURE__ */ jsxRuntime.jsx(
          "rect",
          {
            x: "0.5",
            y: "0.5",
            width: "87",
            height: "87",
            rx: "43.5",
            stroke: "url(#idkit_check_linear)"
          }
        ),
        /* @__PURE__ */ jsxRuntime.jsx("path", { d: "M29.5 45.5L37.5 53.5L57.5 33.5", stroke: "white", strokeWidth: "3" }),
        /* @__PURE__ */ jsxRuntime.jsxs("defs", { children: [
          /* @__PURE__ */ jsxRuntime.jsxs(
            "radialGradient",
            {
              id: "idkit_check_radial",
              cx: "0",
              cy: "0",
              r: "1",
              gradientUnits: "userSpaceOnUse",
              gradientTransform: "translate(20 -1.6729e-06) rotate(63.4349) scale(98.387 97.9627)",
              children: [
                /* @__PURE__ */ jsxRuntime.jsx("stop", { stopColor: "white" }),
                /* @__PURE__ */ jsxRuntime.jsx("stop", { offset: "1", stopColor: "white", stopOpacity: "0" })
              ]
            }
          ),
          /* @__PURE__ */ jsxRuntime.jsxs(
            "linearGradient",
            {
              id: "idkit_check_linear",
              x1: "44",
              y1: "0",
              x2: "44",
              y2: "88",
              gradientUnits: "userSpaceOnUse",
              children: [
                /* @__PURE__ */ jsxRuntime.jsx("stop", { stopColor: "white", stopOpacity: "0.3" }),
                /* @__PURE__ */ jsxRuntime.jsx("stop", { offset: "1", stopColor: "white", stopOpacity: "0" })
              ]
            }
          )
        ] })
      ]
    }
  );
}
function SuccessState() {
  return /* @__PURE__ */ jsxRuntime.jsxs(
    "div",
    {
      style: {
        display: "flex",
        flexDirection: "column",
        alignItems: "center",
        textAlign: "center"
      },
      children: [
        /* @__PURE__ */ jsxRuntime.jsx("div", { className: "idkit-success-icon", children: /* @__PURE__ */ jsxRuntime.jsx(CheckIcon, {}) }),
        /* @__PURE__ */ jsxRuntime.jsx("h2", { className: "idkit-heading", children: __("All set!") }),
        /* @__PURE__ */ jsxRuntime.jsx("p", { className: "idkit-subtext", style: { maxWidth: 260 }, children: __("Your World ID is now connected") })
      ]
    }
  );
}
function ErrorIcon(props) {
  return /* @__PURE__ */ jsxRuntime.jsxs(
    "svg",
    {
      width: "88",
      height: "88",
      viewBox: "0 0 88 88",
      fill: "none",
      xmlns: "http://www.w3.org/2000/svg",
      ...props,
      children: [
        /* @__PURE__ */ jsxRuntime.jsx("rect", { width: "88", height: "88", rx: "44", fill: "#9BA3AE" }),
        /* @__PURE__ */ jsxRuntime.jsx(
          "rect",
          {
            opacity: "0.2",
            width: "88",
            height: "88",
            rx: "44",
            fill: "url(#idkit_error_radial)"
          }
        ),
        /* @__PURE__ */ jsxRuntime.jsx(
          "rect",
          {
            x: "0.5",
            y: "0.5",
            width: "87",
            height: "87",
            rx: "43.5",
            stroke: "url(#idkit_error_linear)"
          }
        ),
        /* @__PURE__ */ jsxRuntime.jsx(
          "path",
          {
            d: "M33.0146 53.9853L43.4999 43.5M53.9851 33.0147L43.4999 43.5M43.4999 43.5L33.0146 33.0147M43.4999 43.5L53.9851 53.9853",
            stroke: "white",
            strokeWidth: "3"
          }
        ),
        /* @__PURE__ */ jsxRuntime.jsxs("defs", { children: [
          /* @__PURE__ */ jsxRuntime.jsxs(
            "radialGradient",
            {
              id: "idkit_error_radial",
              cx: "0",
              cy: "0",
              r: "1",
              gradientUnits: "userSpaceOnUse",
              gradientTransform: "translate(20 -1.6729e-06) rotate(63.4349) scale(98.387 97.9627)",
              children: [
                /* @__PURE__ */ jsxRuntime.jsx("stop", { stopColor: "white" }),
                /* @__PURE__ */ jsxRuntime.jsx("stop", { offset: "1", stopColor: "white", stopOpacity: "0" })
              ]
            }
          ),
          /* @__PURE__ */ jsxRuntime.jsxs(
            "linearGradient",
            {
              id: "idkit_error_linear",
              x1: "44",
              y1: "0",
              x2: "44",
              y2: "88",
              gradientUnits: "userSpaceOnUse",
              children: [
                /* @__PURE__ */ jsxRuntime.jsx("stop", { stopColor: "white", stopOpacity: "0.3" }),
                /* @__PURE__ */ jsxRuntime.jsx("stop", { offset: "1", stopColor: "white", stopOpacity: "0" })
              ]
            }
          )
        ] })
      ]
    }
  );
}
function WarningIcon(props) {
  return /* @__PURE__ */ jsxRuntime.jsxs(
    "svg",
    {
      width: "88",
      height: "88",
      viewBox: "0 0 88 88",
      fill: "none",
      xmlns: "http://www.w3.org/2000/svg",
      ...props,
      children: [
        /* @__PURE__ */ jsxRuntime.jsx("rect", { width: "88", height: "88", rx: "44", fill: "#FFAE00" }),
        /* @__PURE__ */ jsxRuntime.jsx(
          "rect",
          {
            opacity: "0.2",
            width: "88",
            height: "88",
            rx: "44",
            fill: "url(#idkit_warning_radial)"
          }
        ),
        /* @__PURE__ */ jsxRuntime.jsx(
          "rect",
          {
            x: "0.5",
            y: "0.5",
            width: "87",
            height: "87",
            rx: "43.5",
            stroke: "url(#idkit_warning_linear)"
          }
        ),
        /* @__PURE__ */ jsxRuntime.jsx(
          "path",
          {
            d: "M64.1707 59.5415H22.8298L43.4998 22.3354L64.1707 59.5415ZM42.1208 51.3003L42.1218 54.0503H44.8992L44.8982 51.3003H42.1208ZM42.1248 46.7085H44.8748V36.6255H42.1248V46.7085Z",
            fill: "white"
          }
        ),
        /* @__PURE__ */ jsxRuntime.jsxs("defs", { children: [
          /* @__PURE__ */ jsxRuntime.jsxs(
            "radialGradient",
            {
              id: "idkit_warning_radial",
              cx: "0",
              cy: "0",
              r: "1",
              gradientUnits: "userSpaceOnUse",
              gradientTransform: "translate(20 -1.6729e-06) rotate(63.4349) scale(98.387 97.9627)",
              children: [
                /* @__PURE__ */ jsxRuntime.jsx("stop", { stopColor: "white" }),
                /* @__PURE__ */ jsxRuntime.jsx("stop", { offset: "1", stopColor: "white", stopOpacity: "0" })
              ]
            }
          ),
          /* @__PURE__ */ jsxRuntime.jsxs(
            "linearGradient",
            {
              id: "idkit_warning_linear",
              x1: "44",
              y1: "0",
              x2: "44",
              y2: "88",
              gradientUnits: "userSpaceOnUse",
              children: [
                /* @__PURE__ */ jsxRuntime.jsx("stop", { stopColor: "white", stopOpacity: "0.3" }),
                /* @__PURE__ */ jsxRuntime.jsx("stop", { offset: "1", stopColor: "white", stopOpacity: "0" })
              ]
            }
          )
        ] })
      ]
    }
  );
}
var errorCodeVariants = {
  [idkitCore.IDKitErrorCodes.UserRejected]: "cancelled",
  [idkitCore.IDKitErrorCodes.VerificationRejected]: "cancelled",
  [idkitCore.IDKitErrorCodes.FailedByHostApp]: "cancelled",
  [idkitCore.IDKitErrorCodes.Cancelled]: "cancelled",
  [idkitCore.IDKitErrorCodes.ConnectionFailed]: "connection"
};
var variantConfig = {
  cancelled: {
    title: "Request cancelled",
    message: "You've cancelled the request in World App.",
    Icon: WarningIcon
  },
  connection: {
    title: "Connection lost",
    message: "Please check your connection and try again.",
    Icon: ErrorIcon
    // placeholder  swap for WifiOffIcon later
  },
  generic: {
    title: "Something went wrong",
    message: "We couldn't complete your request. Please try again.",
    Icon: ErrorIcon
  }
};
function getVariant(errorCode) {
  if (!errorCode) {
    return "generic";
  }
  return errorCodeVariants[errorCode] ?? "generic";
}
function ErrorState({
  errorCode,
  onRetry
}) {
  const variant = getVariant(errorCode);
  const { title, message, Icon } = variantConfig[variant];
  return /* @__PURE__ */ jsxRuntime.jsxs(
    "div",
    {
      style: {
        display: "flex",
        flexDirection: "column",
        alignItems: "center",
        textAlign: "center"
      },
      children: [
        /* @__PURE__ */ jsxRuntime.jsx("div", { className: "idkit-error-icon", children: /* @__PURE__ */ jsxRuntime.jsx(Icon, {}) }),
        /* @__PURE__ */ jsxRuntime.jsx("p", { className: "idkit-error-title", children: __(title) }),
        /* @__PURE__ */ jsxRuntime.jsx("p", { className: "idkit-error-message", children: __(message) }),
        /* @__PURE__ */ jsxRuntime.jsx("button", { type: "button", className: "idkit-retry-btn", onClick: onRetry, children: __("Try Again") })
      ]
    }
  );
}
function getVisualStage(isSuccess, isError) {
  if (isSuccess) {
    return "success";
  }
  if (isError) {
    return "error";
  }
  return "worldid";
}
function IDKitRequestWidget({
  open,
  onOpenChange,
  onSuccess,
  onError,
  autoClose = true,
  language,
  ...config
}) {
  const flow = useIDKitRequest(config);
  const { open: openFlow, reset: resetFlow } = flow;
  const lastResultRef = react.useRef(null);
  const lastErrorCodeRef = react.useRef(null);
  react.useEffect(() => {
    if (language) {
      setLocalizationConfig({ language });
    }
  }, [language]);
  react.useEffect(() => {
    if (open) {
      openFlow();
      return;
    }
    resetFlow();
  }, [open, openFlow, resetFlow]);
  react.useEffect(() => {
    if (!flow.result || flow.result === lastResultRef.current) {
      return;
    }
    lastResultRef.current = flow.result;
    void Promise.resolve(onSuccess?.(flow.result)).catch(() => {
    });
  }, [onSuccess, flow.result]);
  react.useEffect(() => {
    if (!flow.errorCode || flow.errorCode === lastErrorCodeRef.current) {
      return;
    }
    lastErrorCodeRef.current = flow.errorCode;
    void Promise.resolve(onError?.(flow.errorCode)).catch(() => {
    });
  }, [flow.errorCode, onError]);
  react.useEffect(() => {
    if (flow.isSuccess && autoClose) {
      const timer = setTimeout(() => onOpenChange(false), 2500);
      return () => clearTimeout(timer);
    }
  }, [flow.isSuccess, autoClose, onOpenChange]);
  const stage = getVisualStage(flow.isSuccess, flow.isError);
  const showSimulatorCallout = config.environment === "staging";
  return /* @__PURE__ */ jsxRuntime.jsxs(IDKitModal, { open, onOpenChange, children: [
    stage === "worldid" && /* @__PURE__ */ jsxRuntime.jsx(
      WorldIDState,
      {
        connectorURI: flow.connectorURI,
        isAwaitingUserConfirmation: flow.isAwaitingUserConfirmation,
        showSimulatorCallout
      }
    ),
    stage === "success" && /* @__PURE__ */ jsxRuntime.jsx(SuccessState, {}),
    stage === "error" && /* @__PURE__ */ jsxRuntime.jsx(
      ErrorState,
      {
        errorCode: flow.errorCode,
        onRetry: () => {
          resetFlow();
          openFlow();
        }
      }
    )
  ] });
}

Object.defineProperty(exports, "IDKit", {
  enumerable: true,
  get: function () { return idkitCore.IDKit; }
});
Object.defineProperty(exports, "IDKitErrorCodes", {
  enumerable: true,
  get: function () { return idkitCore.IDKitErrorCodes; }
});
Object.defineProperty(exports, "documentLegacy", {
  enumerable: true,
  get: function () { return idkitCore.documentLegacy; }
});
Object.defineProperty(exports, "orbLegacy", {
  enumerable: true,
  get: function () { return idkitCore.orbLegacy; }
});
Object.defineProperty(exports, "secureDocumentLegacy", {
  enumerable: true,
  get: function () { return idkitCore.secureDocumentLegacy; }
});
Object.defineProperty(exports, "selfieCheck", {
  enumerable: true,
  get: function () { return idkitCore.selfieCheck; }
});
Object.defineProperty(exports, "signRequest", {
  enumerable: true,
  get: function () { return idkitCore.signRequest; }
});
exports.IDKitRequestWidget = IDKitRequestWidget;
exports.useIDKitRequest = useIDKitRequest;
